### PR TITLE
Add starter option in the CLI

### PR DESCRIPTION
**Pull request by Mcastres**
#### My PR is a:

- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix
- [x] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:

- [ ] Admin
- [ ] Documentation
- [ ] Framework
- [ ] Plugin

#### Manual testing done on the following databases:

- [ ] Not applicable
- [ ] MongoDB
- [ ] MySQL
- [ ] Postgres
- [ ] SQLite

### CODE REVIEW

File: packages/strapi-generate-new/lib/create-starter-project.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const execa = require('execa');
+const fs = require('fs-extra');
+const uuid = require('uuid/v4');
```
Comment (alexandrebodin) [inline packages/strapi-generate-new/lib/create-starter-project.js]: can just use uuid directly

File: packages/strapi-generate-new/lib/create-starter-project.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+module.exports = async function createStarterProject(scope) {
+  console.log('Creating a starter project.');
+
+  // Name of the repository
+  const repository = scope.starter;
+  const destination = scope.rootPath;
+  const destinationName = scope.name;
+
+  // Clone the starter repository
```
Comment (alexandrebodin) [inline packages/strapi-generate-new/lib/create-starter-project.js]: You should check if the user has git installed and if the url is a either a git url or an http/https one to dowload the repo differently.

File: packages/strapi-generate-new/lib/create-starter-project.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  });
+
+  // Install dependencies thanks to the package.json inside the starter repo
+  await execa('yarn', ['setup'], {
+    stdio: 'inherit',
+    cwd: destination,
+  });
+
+  // Read the package.json inside the backend directory (strapi)
+  const packageJSON = await fs.readJSON(`${destination}/backend/package.json`);
```
Comment (alexandrebodin) [inline packages/strapi-generate-new/lib/create-starter-project.js]: You should check if the file exists first and through an error with a lodding explaining the starter does'nt match the template

File: packages/strapi-generate-new/lib/create-starter-project.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  // Change the uuid inside the package.json
+  await fs.writeJson(`${destination}/backend/package.json`, {
+    ...packageJSON,
+    strapi: { uuid: uuid() },
+  });
+
+  console.log(`
+  \x1b[32mSuccess!\x1b[0m Cloned ${destinationName}
```
Comment (alexandrebodin) [inline packages/strapi-generate-new/lib/create-starter-project.js]: Use the chalk package for coloring

File: packages/strapi-generate-new/lib/create-starter-project.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  frontend
+
+    \x1b[36mcd ${destinationName}/frontend
+    yarn start\x1b[0m
+      Starts the development server.
+
+  Thanks for giving this starter a try!
+  `);
+
+  if (repository === undefined) {
```
Comment (alexandrebodin) [inline packages/strapi-generate-new/lib/create-starter-project.js]: This check should be done at the begining of the function I think

File: packages/strapi-generate-new/lib/create-starter-project.js (hunk 2 above)
Comment (Mcastres) [reply packages/strapi-generate-new/lib/create-starter-project.js]: I don't understand, do we have to download a git repository differently depending on wether it's `https://github.com/strapi/strapi-starter-react-blog.git` or `git@github.com:strapi/strapi-starter-react-blog.git` ?

Comment (alexandrebodin) [reply packages/strapi-generate-new/lib/create-starter-project.js]: If no git you should download via the archive and unzip it instead of using git clone.

Comment (alexandrebodin) [reply packages/strapi-generate-new/lib/create-starter-project.js]: Also git cloning like that isn't the right way to do it. You need to shallow clone the repo / rm the .git folder and init a new git repo with a clean initial commit.

**Pull request closed without merging**